scrapy/utils/request.py

```diff
@@ -106,3 +107,27 @@ def referer_str(request: Request) -> Optional[str]:
     if referrer is None:
         return referrer
     return to_unicode(referrer, errors='replace')
+
+
+def request_from_dict(d: dict, spider: Optional[Spider] = None) -> Request:
```

Comment:
```suggestion
def request_from_dict(d: dict, *, spider: Optional[Spider] = None) -> Request:
```

Reply:
Done in 1277a7c1385cbec9abce330dcf0e9d4e6ffa8126